<a href="https://colab.research.google.com/github/Sbbarse787/Heart-diseases-predictions-using-Keras-and-pandas/blob/master/Keras_with_Pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive # Mount Colab to Your gdrive
drive.mount('/gdrive')
%cd /gdrive

Drive already mounted at /gdrive; to attempt to forcibly remount, call drive.mount("/gdrive", force_remount=True).
/gdrive


In [ ]:
cd /gdrive/My Drive/Colab Notebooks/SAS  #Specify the Location of your data set

/gdrive/My Drive/Colab Notebooks/SAS


In [ ]:
import pandas as pd
df = pd.read_csv('heart.csv'    #Read the dataset using pandas
)

In [ ]:
df.head()

age  sex  cp  trestbps  chol  fbs  ...  exang  oldpeak  slope  ca  thal  target
0   63    1   3       145   233    1  ...      0      2.3      0   0     1       1
1   37    1   2       130   250    0  ...      0      3.5      0   0     2       1
2   41    0   1       130   204    0  ...      0      1.4      2   0     2       1
3   56    1   1       120   236    0  ...      0      0.8      2   0     2       1
4   57    0   0       120   354    0  ...      1      0.6      2   0     2       1

[5 rows x 14 columns]

In [ ]:
df.isnull().sum()

age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
target      0
dtype: int64

In [ ]:
X = df.iloc[:,0:13]       #Slicing of data set..X=[1,....13]
y = df.iloc[:,-1]          #Y=[Target variable]

In [ ]:
from sklearn.model_selection import train_test_split     #Split the data set
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=0)

In [ ]:
from sklearn.preprocessing import StandardScaler     #Standerisation of dataset
sc = StandardScaler()
sc.fit(X_train)
X_train = sc.transform(X_train)
X_test = sc.transform(X_test)


In [ ]:
pip install --upgrade keras   #install keras

     |████████████████████████████████| 378kB 2.5MB/s 
  Found existing installation: Keras 2.2.5
    Uninstalling Keras-2.2.5:
      Successfully uninstalled Keras-2.2.5


In [ ]:
import keras
from keras.models import Sequential
from keras.layers import Dense
classifier=Sequential()
classifier.add(Dense(units=6,kernel_initializer='uniform',activation='relu',input_dim=13)) # add the layers in neural network..
classifier.add(keras.layers.Dropout(0.4))
classifier.add(Dense(units=6,kernel_initializer='uniform',activation='relu'))
classifier.add(keras.layers.Dropout(0.4))
classifier.add(Dense(units=1,activation='sigmoid'))
classifier.compile(optimizer='adagrad',loss='binary_crossentropy',metrics=['accuracy'])
classifier.fit(X_train,y_train,batch_size=10,epochs=100)


Using TensorFlow backend.


Epoch 1/100
242/242 [==============================] - 0s 1ms/step - loss: 0.6902 - accuracy: 0.5372
Epoch 2/100
242/242 [==============================] - 0s 132us/step - loss: 0.6675 - accuracy: 0.6818
Epoch 3/100
242/242 [==============================] - 0s 120us/step - loss: 0.6387 - accuracy: 0.6446
Epoch 4/100
242/242 [==============================] - 0s 112us/step - loss: 0.6007 - accuracy: 0.7107
Epoch 5/100
242/242 [==============================] - 0s 120us/step - loss: 0.5836 - accuracy: 0.7066
Epoch 6/100
242/242 [==============================] - 0s 119us/step - loss: 0.5488 - accuracy: 0.7479
Epoch 7/100
242/242 [==============================] - 0s 122us/step - loss: 0.5343 - accuracy: 0.7066
Epoch 8/100
242/242 [==============================] - 0s 116us/step - loss: 0.5180 - accuracy: 0.7190
Epoch 9/100
242/242 [==============================] - 0s 123us/step - loss: 0.5098 - accuracy: 0.7314
Epoch 10/100
242/242 [==============================] - 0s 137us/step - los

In [ ]:
df.head(10)

age  sex  cp  trestbps  chol  fbs  ...  exang  oldpeak  slope  ca  thal  target
0   63    1   3       145   233    1  ...      0      2.3      0   0     1       1
1   37    1   2       130   250    0  ...      0      3.5      0   0     2       1
2   41    0   1       130   204    0  ...      0      1.4      2   0     2       1
3   56    1   1       120   236    0  ...      0      0.8      2   0     2       1
4   57    0   0       120   354    0  ...      1      0.6      2   0     2       1
5   57    1   0       140   192    0  ...      0      0.4      1   0     1       1
6   56    0   1       140   294    0  ...      0      1.3      1   0     2       1
7   44    1   1       120   263    0  ...      0      0.0      2   0     3       1
8   52    1   2       172   199    1  ...      0      0.5      2   0     3       1
9   57    1   2       150   168    0  ...      0      1.6      2   0     2       1

[10 rows x 14 columns]

In [ ]:
df.tail(10)

age  sex  cp  trestbps  chol  fbs  ...  exang  oldpeak  slope  ca  thal  target
293   67    1   2       152   212    0  ...      0      0.8      1   0     3       0
294   44    1   0       120   169    0  ...      1      2.8      0   0     1       0
295   63    1   0       140   187    0  ...      1      4.0      2   2     3       0
296   63    0   0       124   197    0  ...      1      0.0      1   0     2       0
297   59    1   0       164   176    1  ...      0      1.0      1   2     1       0
298   57    0   0       140   241    0  ...      1      0.2      1   0     3       0
299   45    1   3       110   264    0  ...      0      1.2      1   0     3       0
300   68    1   0       144   193    1  ...      0      3.4      1   2     3       0
301   57    1   0       130   131    0  ...      1      1.2      1   1     3       0
302   57    0   1       130   236    0  ...      0      0.0      1   1     2       0

[10 rows x 14 columns]

In [ ]:
import numpy as np

In [ ]:
y_pred=classifier.predict(X_test)
y_pred=(y_pred>0.5)
n=[44,1,0,120,169,0,1,144,1,2.8,0,0,1] #Add input of arrays to the model....
prediction=classifier.predict(sc.transform(np.array([n])))  #Prediction of dataset 
prediction=(prediction>0.5)
print(prediction)

[[False]]


In [ ]:
from sklearn.metrics import confusion_matrix
cm=confusion_matrix(y_test,y_pred)
print(cm)

[[22  5]
 [ 3 31]]


In [2]:
def accuracy():
  x=22+31
  y=22+31+5+3
  return (x/y)*100
print(accuracy())  

86.88524590163934


**At the end we get the accuracy of 86.885%**